File: push/src/main/java/com/xiaomi/push/service/PushClientNotifyJob.java
package com.xiaomi.push.service;

import com.xiaomi.channel.commonutils.logger.MyLog;

final class PushClientNotifyJob extends XMPushService.Job {
    String errorType;
    private final PushClientsManager.ClientLoginInfo info;
    int notifyType;
    int reason;
    String reasonMessage;

    PushClientNotifyJob(PushClientsManager.ClientLoginInfo clientLoginInfo) {
        super(0);
        this.info = clientLoginInfo;
    }

    XMPushService.Job build(int i, int i2, String str, String str2) {
        this.notifyType = i;
        this.reason = i2;
        this.errorType = str2;
        this.reasonMessage = str;
        return this;
    }

    @Override
    public String getDesc() {
        return "notify job";
    }

    @Override
    public void process() {
        if (this.info.shouldNotifyClient(this.notifyType, this.reason, this.errorType)) {
            this.info.notifyClientStatus(this.notifyType, this.reason, this.reasonMessage, this.errorType);
            return;
        }
        MyLog.i(" ignore notify client :" + this.info.chid);
    }
}

File: push/src/main/java/com/xiaomi/push/service/PushClientNotifyJob.kt
package com.xiaomi.push.service

import com.xiaomi.channel.commonutils.logger.MyLog

class PushClientNotifyJob(
    private val info: PushClientsManager.ClientLoginInfo,
) : XMPushService.Job(0) {
    var errorType: String? = null
    var notifyType: Int = 0
    var reason: Int = 0
    var reasonMessage: String? = null

    fun build(
        notifyType: Int,
        reason: Int,
        reasonMessage: String?,
        errorType: String?,
    ): XMPushService.Job {
        this.notifyType = notifyType
        this.reason = reason
        this.errorType = errorType
        this.reasonMessage = reasonMessage
        return this
    }

    override fun getDesc(): String = "notify job"

    override fun process() {
        if (info.shouldNotifyClient(notifyType, reason, errorType)) {
            info.notifyClientStatus(notifyType, reason, reasonMessage, errorType)
            return
        }
        MyLog.i(" ignore notify client :${info.chid}")
    }
}
